### DISCUSSION TITLE
Create a UIImage from a DataTemplate

### DISCUSSION THREAD

**Discussion by MarcAntoineHien** [unoplatform/uno#12027] [Q&A] [answered]
Hello ! I was wondering if there is a simple way to take a DataTemplate and create a UIKit.UIImage from it ? 
I need to display a UIImage, but I would like to be able to customize it in my code using a DataTemplate.

Thanks !

**Comment by MarcAntoineHien** [comment]
@kazo0 any idea how I can do that ? Starting from a DataTemplate, I need to convert its content into a Bitmap to then use that Bitmap to create a UIImage. I think this is the way to go, but maybe there is another better way to convert de DataTemplate to a UIImage.

**Reply by kazo0** [reply]
Hmm I might need more details but are you perhaps able to use the RenderTargetBitmap API from WinUI? You can provide a UIElement to generate a bitmap. Not sure about DataTemplates, would depend on what the DataTemplate is being used for

On my phone right now but I can provide more info in a bit

**Reply by jeromelaban** [reply, marked as answer]
@kazo @MarcAntoineHien Indeed, you will not be able to take an image of a `DataTemplate`, but you may be able to use `RenderTargetBitmap` to render the `DataTemplate` in a `ContentControl` using `ContentControl.ContentTemplate`. See [this documentation](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.imaging.rendertargetbitmap?view=winrt-22621) on how to use it.
